app/assets/javascripts/dynamic_search.js

```diff
@@ -0,0 +1,164 @@
+  2)  ESC (KeyCode = 27):  hide typeahead
+  3)  SPACE (KeyCode = 32):  perform dynamic search, do not hide typeahead
+  4)  Any other key:  perform typeahead search, show suggestion list (if not visible)
+**/
+function dynamicSearchKeys(ielem, reslist) {
+	var elemRef = $(ielem);
+	var listelem = $(reslist);
+	elemRef.on("keyup", function(e) {
+		e.preventDefault();
+		var kcode = e.which || e.keyCode;
+		switch(kcode) {
+			case 27:
+				//hit escape, so hide the typeahead
+				listelem.toggleClass('results-noshow',true);
+				break;
+			case 13:
+				//hit enter, so hide typeahead results and run search
+				dynamicSearch(this);
+				listelem.toggleClass('results-noshow',true);
+				break;
+			case 32:
+				//space bar--run search
+				dynamicSearch(this);
+				break;
+			default:
+				//general typing--perform suggestion routine
+				typeaheadSearch(this);
+				break;
+		}
+	});
+}
+
+/**
+  Execute the live search function
+**/
+function dynamicSearch(inputElem) {
+	srchCount += 1;
+        var srchParams = searchParams(inputElem);
+	//don't send nulls or empty strings
+        if (srchParams.srchString && srchParams.srchString != '') {
+	        $.getJSON(srchBase, srchParams, function(sdata) {
+			parseDynamicSearch(sdata);
+		});
+	}
+}
+
+/**
+  Collect up the search parameters and return a search object to be submitted
+**/
+function searchParams(inputElem) {
+  var sparms = {};
+  var stxt = $(inputElem).val();
+  if (stxt) {
+	  sparms.srchString = stxt.trim();
+  }
+  sparms.seq = srchCount;
+  sparms.showCount = 25;
+  sparms.pageNum = 0;
+  return sparms;
+}
+
+/**
+  Parse out the results of the live search and display accordingly
+**/
+function parseDynamicSearch(doclist) {
+	if (doclist) {
+		//if (doclist.seq >= srchCount) {
+			clearResults();
+			//First, display in the "all" tab
+			var currtab = $('#taball');
+			var currcount = $('#allcount');
+			var currres = currtab.find('div.row.sresults-row');
+			var rescon = resultContainer();
+			currcount.text(doclist.items.length);
+			for (var i=0;i<doclist.items.length;i++) {
+				rescon.append($(docPanel(doclist.items[i])));
+			}
+			currres.append(rescon);
+			//Now walk through the array of tabs
+			for (var j=0;j<tabDataArray.length;j++) {
+				var currtype = tabDataArray[j].qryType;
+				var docarray = [];				
+				for (var k=0;k<doclist.items.length;k++) {
+					if (doclist.items[k].docType == currtype) docarray.push(doclist.items[k]);
+				}
+				currtab = $(tabDataArray[j].tabId);
+				currcount = $(tabDataArray[j].countspan);
+				currres = currtab.find('div.row.sresults-row');
+				currcount.text(docarray.length);
+				rescon = resultContainer();
+				for (var m=0;m<docarray.length;m++) {
+					rescon.append($(docPanel(docarray[m])));
+				}
+				currres.append(rescon);
+			}
+		//}
+	}
+}
+
+function resultContainer() {
+	return $("<div class='panel-group'></div>");
+}
+
+/**
+  Create a row result item from the document
+  TODO:  Better way to create these?
```

Comment:
I agree--I just put in the TODO so that it was noted that this is a "manually" created div.

I didn't catch wind of any particular JS or dynamic templating (other than bootstrap layouts), but I was allowing for someone else to have a better idea.